### DISCUSSION TITLE
Preset Saving for Prompts and Settings

### DISCUSSION THREAD

**Discussion by bgeneto** [open-webui/open-webui#6318] [Ideas]
## Feature Request

This feature request proposes the implementation of a preset saving system for prompts and custom settings in Open-webui, similar to the functionality found in LibreChat (see attached image). This would allow users to save and quickly load their preferred prompt configurations, including the system prompt and various advanced parameters.

**Is your feature request related to a problem? Please describe.**

Yes, the lack of a preset saving feature significantly limits the user experience in Open-webui. 

* **Repetitive Input:** Users currently need to manually re-enter their preferred prompts and adjust settings every time they use the application. This is time-consuming and inefficient, especially for complex prompts or frequently used parameter combinations.
* **Limited Experimentation:**  Without the ability to easily save and compare different prompt variations, users might be discouraged from experimenting with different settings and exploring the full potential of the model.

**Describe the solution you'd like:**

I propose implementing a "Save as Preset" button within the chat interface, similar to the one found in LibreChat. This button would:

1. **Capture Current Configuration:** Save the current system prompt, along with all selected advanced parameters (e.g., Temperature, Top K, Context Length) and their values.
2. **Store Preset:**  Store the captured configuration as a named preset, allowing users to provide a descriptive name for easy identification.
3. **Load Preset:**  Provide a mechanism for users to easily browse, select, and load saved presets, instantly applying the stored prompt and settings.

**Describe alternatives you've considered:**

* **Manual Copying and Pasting:**  While users can currently copy and paste prompts and settings, this is a cumbersome workaround and prone to errors. It also lacks the organization and ease of use provided by a dedicated preset system.
* **Waiting for Teams Feature:** While the upcoming Teams feature might offer some form of preset saving, it's unclear if it will address the needs of individual users. Implementing a basic preset system now would provide immediate value to a wider user base.

**Additional context:**

The attached image (see bellow) illustrates a clean and user-friendly implementation of preset saving. A similar approach in Open-webui would greatly enhance the user experience by streamlining workflows, encouraging experimentation, and unlocking the full potential of the platform for all users. 

![feature-request-openwebui](https://github.com/user-attachments/assets/9d8acb7a-9ed4-407e-a132-a90a4dffd16a)

**Comment by jojje** [comment]
Makes perfect sense, since it would allow a use case centric configuration system. I'd argue that most uses of LLMs are use case centric, since the user is using a model for a certain purpose, not to A/B test random queries against different models using the same parameters. The latter is what research, and model tooling creators do, not end users.

For instance. If I need to identify bounding boxes with short object descriptions (annotations) for some specific set of objects in images, I go through the pain of finding a model that has a spatial awareness, is able to recognize my specific objects, then tweak the prompt including system prompt and sampling knobs to get it to output something that's passable for this specific use case. All these choices together end up as "my specific use case configuration for object detection, localization and annotation of cats, dogs, birds ...". Any time I need to perform that same task I have to reset all those parameters exactly.

When I want syntax assist or library function completion in coding, I have to find a suitable model, tweak all the sampling parameters and iterate on a suitable system prompt in order to get a base. Again, a whole set of knobs that must be set to specific values, where the model used is just one out of a bunch of parameters.

This pattern repeats for every single one of all my use cases. The point is that a system prompt isn't disconnected from the model or the other hyper parameters. They all need to be managed together to satisfy _any_ use case, and the original issue the OP opened, that was converted to this discussion topic captures this critical usability gap and suggests a solution that would close the gap.

**Comment by abdessalaam** [comment]
Has there been any update, or workaround, on this?

**Reply by bgeneto** [reply]
Prompts can be saved by users in the Workspace, but they are not "System" prompts. Settings can be applied per model only (AFAIK). So na advances in this feature request (save/load Chat Controls settings) yet :-(

**Comment by mozophe** [comment]
I think that availability of this feature is one of the reasons a lot of users are using LM Studio, even though it's closed source. The ability to save a system prompt and model parameters, which could be used flexibly across different LLMs without needing to clutter the model space, makes it very user friendly as this allows for quickly changing the model parameters and system prompt, for different use cases.

Is this feature planned?

**Comment by dancingmadkefka** [comment]
I am just discovering the sad reality that this feature does not exist. I'm shocked, given how popular this tool seems to be. This will definitely limit my usage of it. The UX could generally really do with an overhaul. LM Studio is really intuitive and powerful in comparison.

Even the selecting of parameters and writing of system prompt is frustrating due to the obnoxious tooltips blocking the parameters and the system prompt window being too small an not resizable (off-topic I know sorry).

Will be looking out for this feature in the future :)

**Comment by disarticulate** [comment]
Certainly seems like a bizarre omission. A monkey fix is to save it to the model:<img width="797" height="425" alt="image" src="https://github.com/user-attachments/assets/93b74b4b-94f4-41d0-bbbb-46d7fe3c0e42" />

Found under Admin -> Settings.

Obviously this doesn't work for per user prompts.

**Comment by MichaelTScott** [comment]
bump, 

I assume this still doesn't exist? Model behavior can vary wildly with different "Advanced parameters" and system prompts. 

Similar behavior can be achieved but duplicating the model but imo this is less clean

**Comment by Leseratte10** [comment]
I'd be interested in this feature as well after changing from LM Studio to Open WebUI. I regularly experiment with different system prompts in different chats. Other UIs keep the configured system prompt stored separately for each chat, so when I go back to an old chat and add a message, the model still knows the original system prompt. And I can look it up, if needed. 

In Open WebUI, it looks like all the settings I can do by clicking the settings icon in the top right corner, including a custom prompt, aren't saved at all, neither globally nor per chat, so as soon as I move away from a chat they get cleared, and when I get back to a chat later I have no idea what system prompt this particular chat used.

I played around with making a custom model for my system prompts, but A) I often want to try the same system prompt with different models, and B) this would mess up all the internal statistics on model usage if I need to make a new model for each prompt.

First I assumed that the prompts under "Workspace" would be what I'm looking for, but these just seem to be shortcuts to append boilerplate text in a user message and can't be used as system prompts either.
